sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -149,14 +289,14 @@ static ProcessElementMethod create(
 
     /** Whether this {@link DoFn} uses a Single Window. */
     public boolean usesSingleWindow() {
-      return extraParameters().contains(Parameter.BOUNDED_WINDOW);
+      return extraParameters().contains(Parameter.boundedWindow());
```

Comment:
As mentioned, I feel differently about this. I agree that it is important to know that all `BoundedWindowParameter` descriptors are equal, but it is not important that they are all one object. That latter is a minor allocation savings (probably zero since we don't touch them in a hot loop) which I have implemented, but doesn't require API asymmetry or abstraction puncturing.

While the code was already correct from that perspective, I agree that it is more readable to be more explicit. So I have switched back to `AutoValue` to make the actual implementation foolproof and future-proof, while adding the spec to the Javadoc.